source/xml_parser.c

```diff
@@ -300,8 +307,12 @@ int aws_xml_node_traverse(
             parent_closed = true;
         }
 
-        size_t node_name_len = end_location - next_location;
-
+        size_t node_name_len;
+        if (aws_sub_size_checked((size_t)end_location, (size_t)next_location, &node_name_len)) {
```

Comment:
maybe use an if check against the pointer, which will be safer

Reply:
Thanks, updated.